### PR TITLE
Feature/fix serialization and storage status v2

### PR DESCRIPTION
**Pull request by vedrani**

### CODE REVIEW

File: src/serialization/toSerializableFormat.js [hunk 1]
```diff
@@ -32,15 +32,18 @@ function saveStatus(originalState, serializableState) {
  * @returns {*}
  */
 function transformSubstate(originalSubState) {
-  if (_.isPlainObject(originalSubState)) {
-    // eslint-disable-next-line  no-use-before-define
-    return toSerializableFormat(originalSubState);
-  } else if (_.isArray(originalSubState) && originalSubState[STATUS]) {
+  if (_.isArray(originalSubState)) {
+    if (!originalSubState[STATUS]) {
```
Comment (MrBr) [inline src/serialization/toSerializableFormat.js]: Why have you changed the condition? You have nested `if` in the `if`.

Comment (vedrani) [inline src/serialization/toSerializableFormat.js]: I need ordinary serialization if it-s ordianry array so this second if is break condition before running arrayToObject logic.

Comment (MrBr) [inline src/serialization/toSerializableFormat.js]: I still do not understand, you can have the condition in one line`_.isArray(originalSubState) && originalSubState[STATUS]`? If this is not true, `originalSubState` is going to be returned.

`_. isPlainObject([])` will return `false`.

Comment (vedrani) [inline src/serialization/toSerializableFormat.js]: Agree, will test and fix. I used `isObjectLike` so maybe that is the reason why I left it.  Thanks.

**Pull request merged**